src/Pact/Analyze/Eval/Term.hs

```diff
@@ -288,6 +290,32 @@ readFields tn sRk tid (SObjectUnsafe (SCons' sym fieldType subSchema)) = do
     AVal _ _ -> error
       "impossible: unexpected type of cell provenance in readFields"
 
+mkChainData :: SingTy ('TyObject ty) -> Analyze (S (ConcreteObj ty))
+mkChainData (SObjectUnsafe SNil') = pure $ sansProv $ literal ()
+mkChainData (SObjectUnsafe (SCons' sym (fieldType :: SingTy v) subSchema)) = do
+  let fieldName = symbolVal sym
+      subObjTy = SObjectUnsafe subSchema
+      sbv = withHasKind fieldType $ uninterpret $ "chain_data_" <>
+        (map (\c -> if c == '-' then '_' else c) fieldName)
+
+      mNumDict :: Maybe (Dict (Num (Concrete v)))
+      mNumDict = case fieldType of
+        SInteger -> Just Dict
+        SDecimal -> Just Dict
+        _        -> Nothing
+
+  withSymVal fieldType $ do
+    when (any (fieldName ==) ["block-height", "gas-limit", "gas-price"]) $ do
```

Comment:
nice :+1:.

I find this a bit more readable: 

```
when (fieldName `elem` ["block-height", "gas-limit", "gas-price"])
```

Reply:
Agreed -- that's better